drivers/audio/max98091.c

```diff
@@ -0,0 +1,354 @@
+
+	max98091_out_volume_config(dev, AUDIO_CHANNEL_ALL, M98091_DEFAULT_VOLUME);
+	max98091_out_mute_config(dev, AUDIO_CHANNEL_ALL, false);
+}
+
+static void max98091_start_output(const struct device *dev)
+{
+}
+
+static void max98091_stop_output(const struct device *dev)
+{
+}
+
+static int max98091_set_property(const struct device *dev, audio_property_t property,
+				 audio_channel_t channel, audio_property_value_t val)
+{
+	uint8_t reg, mask;
+
+	switch (property) {
+	case AUDIO_PROPERTY_OUTPUT_VOLUME:
+		max98091_out_volume_config(dev, channel, val.vol);
+	case AUDIO_PROPERTY_OUTPUT_MUTE:
+		max98091_out_mute_config(dev, channel, val.mute);
+	default:
+		return -EINVAL;
+	}
+}
+
+static int max98091_configure(const struct device *dev, struct audio_codec_cfg *cfg)
+{
+	const struct max98091_config *const dev_cfg = DEV_CFG(dev);
+
+	if (cfg->dai_type >= AUDIO_DAI_TYPE_INVALID) {
+		LOG_ERR("dai_type not supported");
+		return -EINVAL;
+	}
+
+	max98091_soft_reset(dev);
+
+	if (cfg->dai_route == AUDIO_ROUTE_BYPASS) {
+		return 0;
+	}
+
+	/* Put the audio codec into shutdown mode */
+	max98091_write_reg(dev, M98091_REG_DEVICE_SHUTDOWN, 0x00);
+
+	max98091_write_reg(dev, M98091_REG_DAC_CONTROL, 0x00);
+
+	max98091_write_reg(dev, M98091_REG_TDM_CONTROL, 0x00);
+
+	/* Set DLY = 1 to conform to the I2S standard.
+	 * DLY is only effective when TDM = 0
+	 */
+	max98091_write_reg(dev, M98091_REG_INTERFACE_FORMAT, M98091_DLY_MASK);
+
+	/* Configure system clock */
+	max98091_set_system_clock(dev, dev_cfg->mclk_freq);
+
+	max98091_protocol_config(dev, cfg->dai_type);
+	max98091_audio_fmt_config(dev, &cfg->dai_cfg);
+
+	/* Configure audio paths based on route */
+	switch (cfg->dai_route) {
+	case AUDIO_ROUTE_PLAYBACK:
+		max98091_configure_output(dev);
+		break;
+	default:
+		LOG_DBG("Unsupported audio route selected");
+		break;
+	}
+
+	/* Bring the audio codec out of shutdown mode */
+	max98091_write_reg(dev, M98091_REG_DEVICE_SHUTDOWN, M98091_SHDNN_MASK);
+
+	return 0;
+}
+
+static const struct audio_codec_api max98091_api = {
+	.configure = max98091_configure,
+	.start_output = max98091_start_output,
+	.stop_output = max98091_stop_output,
+	.set_property = max98091_set_property,
+};
+
+static int max98091_init(const struct device *dev)
+{
+	const struct max98091_config *cfg_tan = dev->config;
+	uint8_t device_id;
+
+	if (!device_is_ready(cfg_tan->i2c.bus)) {
+		LOG_ERR("I2C bus not ready");
+		return -ENODEV;
+	}
+
+	max98091_read_reg(dev, M98091_REG_REVISION_ID, &device_id);
+	if (device_id >= M98091_REVA && (device_id <= M98091_REVA + 0x0f)) {
+		LOG_INF("MAX98091 Device ID: 0x%02X", device_id);
+		return 0;
+	}
+
+	LOG_INF("Invalid MAX98091 Correct Device ID: 0x%02X", device_id);
+	return -ENODATA;
+}
+
+#define MAX98091_INIT(inst)								\
+	static const struct max98091_config max98091_config_##inst = {			\
+		.i2c = I2C_DT_SPEC_INST_GET(inst),					\
+		.mclk_freq = DT_INST_PROP_OR(inst, mclk_frequency, 0)};			\
```

Comment:
The property is required

```suggestion
		.mclk_freq = DT_INST_PROP(inst, mclk_frequency)};			\
```